### PR TITLE
Add COLNEW benchmark

### PR DESCRIPTION
**Pull request by ErikQQY**
Add benchmarks for COLNEW method

### CODE REVIEW

File: benchmarks/NonlinearProblem/nonlinear_solver_23_tests.jmd [hunk 1]
```diff
@@ -565,7 +565,6 @@ end
 save("summary_wp_23test_problems.svg", fig)
 ```
 
-# Appendix
```
Comment (ChrisRackauckas) [inline benchmarks/NonlinearProblem/nonlinear_solver_23_tests.jmd:568]: Revert this? I don't understand why we would remove the standard appendix header

Comment (ErikQQY) [reply benchmarks/NonlinearProblem/nonlinear_solver_23_tests.jmd:568]: This one is duplicated, see:
https://github.com/SciML/SciMLBenchmarks.jl/blob/edcbe452deb980f9379204344bc6e525067cd846/src/SciMLBenchmarks.jl#L89-L96

And

https://docs.sciml.ai/SciMLBenchmarksOutput/dev/NonlinearProblem/nonlinear_solver_23_tests/#Appendix

Comment (ChrisRackauckas) [reply benchmarks/NonlinearProblem/nonlinear_solver_23_tests.jmd:568]: gotcha

**Comment by ChrisRackauckas** [comment]
This should get rerun

**Pull request merged**